src/tiled/editableworld.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * editableworld.h
+ * Copyright 2023, Chris Boehm AKA dogboydog
+ * Copyright 2023, Thorbjørn Lindeijer <bjorn@lindeijer.nl>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "editableasset.h"
+#include "worlddocument.h"
+#include "worldmanager.h"
+
+#include <QObject>
+
+namespace Tiled {
+
+/**
+ * Wrapper which allows world structs to be used with the EditableAsset
+ * class.
+ */
+class ScriptWorld : public Object
```

Comment:
I wonder if not the `World` class itself should derive from `Object` instead.

Reply:
We could do that if you think it's best, I just wasn't sure if you would want me to change the base definition of Worlds to support scripting